src/packagedcode/pypi.py

```diff
@@ -303,6 +305,33 @@ def parse_setup_py(location):
     )
 
 
+def parse_requirements_txt(location):
+    """
+    Return a Package built from requirements.txt.
+    """
+    dependencies = OrderedDict()
+    with io.open(location, encoding='utf-8') as loc:
+        for req in requirements.parse(loc):
+            dependencies[req.name] = req.specs
+    package_dependencies = []
+    for dep_name, dep_specs in dependencies.items():
```

Comment:
What benefits do you gain from reading your requirements in the `dependencies` mapping and then reding that mapping `dependencies.items()` here?